### ISSUE TITLE
`<div>` element is not permitted as content under `<code>`

### ISSUE THREAD

**Issue by selfire1** [nuxt/content#2007]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.10.0`
- Nuxt Version:     `3.4.0`
- Nitro Version:    `2.3.3`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `extends`, `modules`, `runtimeConfig`, `app`, `image`, `content`, `nitro`
- Runtime Modules:  `nuxt-icon@0.3.3`, `@nuxt/content@2.5.2`, `@nuxt/image-edge@1.0.0-28020728.5df24eb`, `@nuxtjs/tailwindcss@6.6.5`, `@nuxtjs/html-validator@1.2.4`, `@nuxtjs/color-mode@3.2.0`
- Build Modules:    `-`

### Reproduction

1. Enable `highlight` in the content config

```vue
export default defineNuxtConfig({
  modules: ['@nuxt/content'],
  content: {
    highlight: {
      // https://github.com/shikijs/shiki/blob/main/docs/themes.md
      theme:  'min-light',
    },
  },
});
```

2. Render a fenced code block.

---

* Minimal reproduction on StackBlitz: [StackBlitz](https://stackblitz.com/edit/github-fwhuqq?file=nuxt.config.ts)

### Describe the bug

This is an example of how Content outputs a fenced code block with `highlight` enabled (Example taken from the NuxtContent Website: [queryContent() · Nuxt Content](https://content.nuxtjs.org/api/composables/query-content))

```html
<pre>
  <code>
    <div class="line"><span class="ct-952581">// Create a query looking for anything in content/ directory</span></div>
    <div class="line"><span class="ct-d7952f">const</span><span class="ct-cf3e01"> </span><span
        class="ct-d584cd">contentQuery</span><span class="ct-cf3e01"> </span><span class="ct-d7952f">=</span><span
        class="ct-cf3e01"> </span><span class="ct-b07143">queryContent</span><span class="ct-cf3e01">()</span></div>
    <div class="line"></div>
    <!-- … -->
  </code>
</pre>
```

Using [nuxtjs/html-validator](https://html-validator.nuxtjs.org/), this throws an error for me: `error  <div> element is not permitted as content under <code>   element-permitted-content`.

I checked the MDN docs and in fact, `<code>` [only accepts **phrasing content**](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/code#technical_summary) as child content. In return, permitted parents for `<div>` are [any elements that accept **flow content**](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div#technical_summary).

If I understand correctly, nesting `<div>` under `<code>` is invalid HTML. ~~Hence, this Content's DOM output results in Hydration Mismatches.~~ Edit: While actually there are no Hydration Mismatches thrown, this rendering still isn't valid HTML and it might be worth looking into other solutions.

### Additional context

* [Shiki's website](https://shiki.matsu.io/) uses span elements itself to highlight tokens in code blocks.
* Related PR that changes `span` to `div` #1885 to fix this issue #1883
* Docusaurus resolved the Firefox issue by using `<br>` tags for line breaks, maybe that's something worth investigating? (https://github.com/facebook/docusaurus/pull/5668)

**Comment by nobkd** [comment]
Can be closed, as it is changed in edge channel.

**Comment by farnabaz** [comment]
Resolved in #2008, and will be released in v2.6
As @nobkd said, for now you can use [edge-channel](https://content.nuxtjs.org/guide/migration/edge-channel) for the fix
